### PR TITLE
feat: support options change

### PR DESCRIPTION
**Pull request by FRSgit**
follow @floating-ui/react-dom api
make update method available refactor implementation
clean up the typings
fix eslint configuration for packages/dom
configure pnpm to install peerdeps locally
remove global workspace deps
write global scripts for starting app & linting
remove unnecessary pnpm-lock.yml from apps/example refactor example a bit
add missing deps to packages/dom
remove dangling commas from tsconfigs' to make then JSON-compatible

Signed-off-by: Jakub Freisler <jakub@frsource.org>

### CODE REVIEW

**Comment by FRSgit** [comment]
@Szymon-dziewonski please review commits separately - the second one is just a result of running `pnpm lint:fix` on whole repo

File: .npmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+auto-install-peers=true
```
Comment (Szymon-dziewonski) [inline .npmrc:1]: thats nice ! is it working as well with yarn?

File: apps/example/src/components/HelloWorld.vue [hunk 2]
```diff
@@ -13,21 +12,11 @@ const { x, y, reference, floating, strategy } = useFloating({
...
+  <h5>Tooltip position:</h5>
+  <p>x: {{ x }} y: {{ y }}</p>
+  <button ref="reference">Button</button>
   <div
-    :ref="
-      (el) => {
-        floating = el;
-      }
-    "
+    ref="floating"
```
Comment (Szymon-dziewonski) [inline apps/example/src/components/HelloWorld.vue:19]: its working like that ?

File: apps/example/vite.config.ts [hunk 3]
```diff
@@ -5,6 +5,11 @@ import vue from '@vitejs/plugin-vue'
 
 // https://vitejs.dev/config/
 export default defineConfig({
+  server: {
+    watch: {
+      ignored: ['!**/node_modules/@frsource/floating-ui-vue-dom/**']
```
Comment (Szymon-dziewonski) [inline apps/example/vite.config.ts]: <3

File: packages/dom/src/index.ts [hunk 4]
```diff
@@ -1,56 +1,57 @@
...
+      floatingInstance.value = await createFloatingUIInstance(unrefElement(referenceRef), unrefElement(floatingRef), optionsInternal);
+    },
+    {immediate: true}
+  );
 
   onScopeDispose(() => floatingInstance.value?.cleanup());
 
   return {
     cleanup: computed(() => floatingInstance.value?.cleanup),
+    update: computed(() => floatingInstance.value?.update),
```
Comment (Szymon-dziewonski) [inline packages/dom/src/index.ts:47]: noice!

**Review by Szymon-dziewonski — APPROVED** [review]

File: apps/example/src/components/HelloWorld.vue (hunk 2 above)
Comment (FRSgit) [reply apps/example/src/components/HelloWorld.vue:19]: yup

File: apps/example/vite.config.ts (hunk 3 above)
Comment (FRSgit) [reply apps/example/vite.config.ts]: yea, that's so cool 😄

**Pull request merged**